crates/nu-command/src/dataframe/values/nu_expression/mod.rs

```diff
@@ -174,16 +174,6 @@ pub fn expr_to_value(expr: &Expr, span: Span) -> Value {
     let cols = vec!["expr".to_string(), "value".to_string()];
 
     match expr {
-        Expr::Not(expr) => {
```

Comment:
It seems like we're losing several expressions throughout this file. Is that because they don't exist anymore in polars or some other reason? I also wonder if there's a different way to do this now, since I'm assuming they just don't exist anymore in the Expr enum.

Reply:
Polar moves these expr variant from `Expr` into [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html).

And I just take a look about these variant, it seems that we actually have no way to construct these removed `Expr`.

Reply:
And after take a closer look about NuExpression's constructors(from nushell value): try_from_value.

It seems that Only specific variants' of polar expression can be constructed

Reply:
Ah! That's a good catch!  I need to take a closer look around NuExpression with your question.

Reply:
Sorry for waiting for a long time, here is my investigation result:

For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr]`https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html`, and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

Reply:
For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html), and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

Reply:
Sorry for a relative long time, here is my investigation result

For the removing, we need to know how `Expr` is constructed from nushell's Value or pipeline input.  

And it's constructed by `NuExpression::try_from_value` and `NuExpression::try_from_pipeline`(which finally leads to `NuExpression::try_from_value`).  In general, nushell takes user argument input, and convert it to `NuExpression` through `try_from_value` method.

After looking into `try_from_value`, it seems that it only accepts `String, Int, Bool, Float` as input.  So no `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` is used.  So the reason they exists is satify our compilder, and don't need to use something like `expect`(to expect these expression never occurred), and we don't want to throws out error.

Underhood, `Expr::Not`, `Expr::IsNotNull`, `Expr::IsNull` are removed from `Expr` to [FunctionExpr](https://docs.rs/polars/latest/polars/prelude/enum.FunctionExpr.html), and I've checked `FunctionExpr` usage in nushel, I think we don't use them in source code.  So it's ok to just take and go through with it.

> Is that because they don't exist anymore in polars or some other reason?

So as conclusion: I think they exists just for satifying our compiler, and current version of nushell never execute it